src/Form/FormAnalyzer.js

```diff
@@ -158,7 +158,11 @@ class FormAnalyzer {
         // check button contents
         if (el.matches(SUBMIT_BUTTON_SELECTOR)) {
             // If we're sure this is a submit button, it's a stronger signal
-            const strength = el.getAttribute('type') === 'submit' ? 20 : 2
+            const strength =
+                el.getAttribute('type') === 'submit' ||
+                /primary|submit/i.test(el.className) ||
+                el.offsetHeight * el.offsetWidth >= 10000
```

Comment:
Here we try to give a more strong signal when the button we're evaluating is large (250x40 px). The measure is somewhat arbitrary, but the idea is that if a **big** button says "Sign in" we want to give a strong signal towards it being a sign in form.